### PR TITLE
List keyboard behavior test and fix lint issues

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
Modifies one of the list tests to use the 'space' key on the onOrder buttons. The test covers behavior added in https://github.com/grommet/grommet/pull/7140.

https://github.com/grommet/grommet/pull/7140 introduced a lint error because the dependency array for onSelectOption didn't include everything. I ended up removing the `useCallback` for that function. I don't think we were getting much value from `useCallback` in this case.

I modified our lint rules to flag react-hooks/exhaustive-deps as an error instead of just a warning. I was not able to find a way to have lint fail on warnings (but I didn't spend a ton of time looking, so I could have missed something).

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
no
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/List/__tests__/List-test.js [hunk 1]
```diff
@@ -581,7 +581,7 @@ describe('List onOrder', () => {
     expect(container.firstChild).toMatchSnapshot();
   });
 
-  test('Keyboard move up', () => {
+  test.only('Keyboard move up', () => {
```
Comment (jcfilben) [inline src/js/components/List/__tests__/List-test.js:584]:
```suggestion
  test('Keyboard move up', () => {
```

File: src/js/components/List/__tests__/List-test.js [hunk 2]
```diff
@@ -600,11 +600,11 @@ describe('List onOrder', () => {
     // beta's up arrow control should be active
     expect(container.firstChild).toMatchSnapshot();
     fireEvent.keyDown(getByText('alpha'), {
-      key: 'Enter',
-      keyCode: 13,
-      which: 13,
+      key: 'Space',
```
Comment (taysea) [inline src/js/components/List/__tests__/List-test.js:603]: Did we need to change this test? Seems like `Enter` is supported

File: src/js/components/List/List.js [hunk 3]
```diff
@@ -252,54 +245,41 @@ const List = React.forwardRef(
...
-            onClickItem(adjustedEvent);
-            sendAnalytics({
-              type: 'listItemClick',
-              element: listRef.current,
-              event: adjustedEvent,
-              item: data[active],
-              index: active,
-            });
-          }
+    const onSelectOption = (event) => {
```
Comment (taysea) [inline src/js/components/List/List.js:248]: Pausing about removing `useCallback` vs just adding the needed dependencies. Reading docs about function caching, is there value to keeping it if it wasn't causing bugs?

https://react.dev/reference/react/useCallback

File: .eslintrc [hunk 4]
```diff
@@ -64,7 +64,7 @@
     "react/sort-comp": 0,
     "react/state-in-constructor": 0,
     "react/static-property-placement": 0,
-    "react-hooks/exhaustive-deps": 1,
+    "react-hooks/exhaustive-deps": 2,
```
Comment (taysea) [inline .eslintrc:67]: I'm good with this.

File: src/js/components/List/__tests__/List-test.js (hunk 2 above)
Comment (britt6612) [reply src/js/components/List/__tests__/List-test.js:603]: nice ☝️ thinking

**Review by taysea — APPROVED** [review]
Looks good!

**Review by britt6612 — APPROVED** [review]

**Pull request merged**